packages/extension/src/background/wallet.ts

```diff
@@ -0,0 +1,275 @@
+      compileCalldata({ signer: starkPub, guardian: "0" }),
+      seed,
+    )
+
+    // TODO: register a L1 address with the wallet as soon as some registry is online
+
+    if (
+      deployTransaction.code !== "TRANSACTION_RECEIVED" ||
+      !deployTransaction.address
+    ) {
+      throw new Error("Deploy transaction failed")
+    }
+
+    const account = {
+      network: networkId,
+      address: deployTransaction.address,
+      signer: {
+        type: "local_secret",
+        derivation_path: getPathForIndex(index),
+      },
+    }
+
+    this.accounts.push(account)
+
+    await this.writeBackup()
+    await this.selectAccount(deployTransaction.address)
+
+    return { account, txHash: deployTransaction.transaction_hash }
+  }
+
+  public getAccounts(): WalletAccount[] {
+    return this.accounts
+  }
+
+  public async getSelectedAccountSigner(): Promise<Signer> {
+    if (!this.isSessionOpen()) {
+      throw Error("no open session")
+    }
+
+    const account = await this.getSelectedAccount()
+
+    const keyPair = getStarkPair(
+      account.signer.derivation_path,
+      this.session?.secret as string,
+    )
+    const provider = getProvider(account.network)
+    return new Signer(provider, account.address, keyPair)
+  }
+
+  public async getSelectedAccount(): Promise<WalletAccount> {
+    if (this.accounts.length === 0) {
+      throw new Error("no accounts")
+    }
+
+    const address = await this.store.getItem("SELECTED")
+    const account = this.accounts.find((account) => account.address === address)
+    return account ?? this.accounts[0]
+  }
+
+  public async selectAccount(address: string) {
+    const account = this.accounts.find((account) => account.address === address)
+    if (account === undefined) {
+      return
+    }
+    await this.store.setItem("SELECTED", account.address)
+  }
+
+  public async removeAccount(address: string) {
+    if (!this.isSessionOpen()) {
+      throw Error("no open session")
+    }
+    this.accounts = this.accounts.filter(
+      (account) => account.address !== address,
+    )
+    await this.writeBackup()
+  }
+
+  public lock() {
+    this.session = undefined
+  }
+
+  public async importBackup(backupString: string) {
+    if (!Wallet.validateBackup(backupString)) {
+      throw new Error("invalid backup file")
+    }
+    await this.store.setItem("BACKUP", backupString)
+    await this.setup()
+  }
+
+  public exportBackup(): { url: string; filename: string } {
+    if (!this.isInitialized()) {
+      throw Error("no local backup")
+    }
+    const blob = new Blob([this.encryptedBackup as string], {
+      type: "application/json",
+    })
+    const url = URL.createObjectURL(blob)
+    const filename = "argent-x-backup.json"
+    return { url, filename }
+  }
+
+  public static validateBackup(backupString: string): boolean {
+    try {
+      const backup = JSON.parse(backupString)
+      return ajv.validate(backupSchema, backup)
+    } catch (error) {
```

Comment:
can remove error variable if unused

```suggestion
    } catch {
```